sdk/storage/Azure.Storage.Blobs/README.md

```diff
@@ -33,77 +44,131 @@ Blob storage is designed for:
 
-string marker;
-do
+// Get a reference to a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, "sample-container");
+
+// List all of its blobs
+foreach (BlobItem blob in container.GetBlobs())
 {
-    var response = await container.ListBlobsFlatSegmentAsync(marker);
-    foreach (var blob in response.Value.BlobItems)
-    {
-        Console.WriteLine(blob.Name);
-    }
-    marker = response.Value.NextMarker;
+    Console.WriteLine(blob.Name);
 }
-while (!string.IsNullOrEmpty(marker));
 ```
 
+### Authenticating with Azure.Identity
+
+The [Azure Identity library](identity) provides easy Azure Active Directory support for authentication.
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+// Create a BlobServiceClient that will authenticate through Active Directory
+Uri accountUri = new Uri("https://MYSTORAGEACCOUNT.blob.core.windows.net/");
+BlobServiceClient client = new BlobServiceClient(accountUri, new DefaultAzureCredential());
+```
+
+Learn more about enabling Azure Active Directory for authentication with Azure Storage in [our documentation](storage_ad) and \[our samples](COMING TODAY).
+
 ## Troubleshooting
+
 All Blob service operations will throw a
 [StorageRequestFailedException][StorageRequestFailedException] on failure with
-helpful [`ErrorCode`s][error_codes].
+helpful [`ErrorCode`s][error_codes].  Many of these errors are recoverable.
+
+```c#
+string connectionString = "<connection_string>";
+
+// Try to create a container named "sample-container"
+BlobContainerClient container = new BlobContainerClient(connectionString, Randomize("sample-container"));
+try
+{
+    container.Create();
+}
+catch (StorageRequestFailedException ex)
+    when (ex.ErrorCode == BlobErrorCode.ContainerAlreadyExists)
+{
+    // Ignore any errors if the container already exists
+}
+```
 
 ## Next steps
-Get started with our [Blob samples][samples].
+
+Get started with our [Blob samples][samples]:
+
+1. [Hello World](samples/Sample01a_HelloWorld.cs): Upload, download, and list blobs (or [asynchronously](samples/Sample01b_HelloWorldAsync.cs))
 
 ## Contributing
+
 This project welcomes contributions and suggestions.  Most contributions require
 you to agree to a Contributor License Agreement (CLA) declaring that you have
 the right to, and actually do, grant us the rights to use your contribution. For
-details, visit https://cla.microsoft.com.
+details, visit [cla.microsoft.com][cla].
 
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/)
-or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any
+This project has adopted the [Microsoft Open Source Code of Conduct][coc].
+For more information see the [Code of Conduct FAQ][coc_faq]
+or contact [opencode@microsoft.com][coc_contact] with any

```

Comment:
I *think* md supports defining text along with URL for named URLs.  Might reduce some of the bookkeeping here.

Reply:
It does, but the initial template pulled them all out to the bottom and the team has been sticking with that so I made it consistent.